### PR TITLE
Create and send data requests

### PR DESCRIPTION
**Pull request by anler**
## What

Endpoints for creating and sending data requests.

### `/createDataRequest`

Request body:

```json
{
  "sessionId": "<session id>",
  "walletId": "<wallet id>",
  "label": "<optional label>",
  "fee": "<numeric value>",
  "request": {
    "dataRequest": {
      "timeLock": "<numeric value>",
      "retrieve": ["<RAD Retrieve object"],
      "aggregate": "RAD Aggregate object",
      "tally": "RAD Tally object"
    },
    "value": "<numeric value>",
    "witnesses": "<numeric value>",
    "backupWitnesses": "<numeric value>",
    "commitFee": "<numeric value>",
    "revealFee": "<numeric value>",
    "tallyFee": "<numeric value>"
  }
}
```

Response body:

```json
{
  "transactionId": "<created transaction id>",
  "transaction": "json-serialized transaction",
  "bytes": "hex-encoded transaction",
}
```

### `/sendDataRequest`

Request body

```json
{
  "sessionId": "<session id>",
  "walletId": "<wallet id>",
  "transactionId": "<transaction id>"
}
```

### CODE REVIEW

File: wallet/src/types.rs [hunk 1]
```diff
@@ -113,17 +116,46 @@ pub struct VttParams {
...
+
+#[derive(Debug)]
+pub struct Success;
+
+impl serde::Serialize for Success {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str("success")
```
Comment (tmpolaczyk) [inline wallet/src/types.rs]:
```suggestion
        serializer.serialize_bool(true)
```

Comment (anler) [reply wallet/src/types.rs]: image

File: wallet/src/actors/app/handlers/create_data_req.rs [hunk 2]
```diff
@@ -3,21 +3,107 @@ use serde::{Deserialize, Serialize};
...
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Debug, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct CreateDataReqRequest {
-    pub rad_request: types::RADRequest,
+    session_id: types::SessionId,
+    wallet_id: String,
+    label: Option<String>,
+    request: DataRequestOutput,
+}
```
Comment (tmpolaczyk) [inline wallet/src/actors/app/handlers/create_data_req.rs]: We also need a `data_request_fee`, which is the fee for including the `DataRequestTransaction` into a block. The cost for creating a data request is `data_request.value + fee`.

```suggestion
    fee: u64
}
```

File: wallet/src/repository/wallet/mod.rs [hunk 3]
```diff
@@ -380,16 +382,111 @@ where
...
+    }
+
+    /// Create a new data request transaction using available UTXOs.
+    pub fn create_data_req(
+        &self,
+        types::DataReqParams { label, request }: types::DataReqParams,
+    ) -> Result<types::DRTransaction> {
+        let fee = request.fee();
+        let value = request.total_witnesses_reward();
+        eprintln!("REQUEST: {:?}", request);
```
Comment (tmpolaczyk) [inline wallet/src/repository/wallet/mod.rs]: Change eprintln to log or remove it.

```suggestion
        log::debug!("REQUEST: {:?}", request);
```

File: wallet/src/actors/app/handlers/create_data_req.rs [hunk 4]
```diff
@@ -3,21 +3,107 @@ use serde::{Deserialize, Serialize};
...
+#[serde(rename_all = "camelCase")]
+struct DataRequestOutput {
+    data_request: RADRequest,
+    value: u64,
+    witnesses: u16,
+    backup_witnesses: u16,
+    commit_fee: u64,
+    reveal_fee: u64,
+    tally_fee: u64,
+    time_lock: u64,
```
Comment (tmpolaczyk) [inline wallet/src/actors/app/handlers/create_data_req.rs]: Can you move this field to inside `RadRequest`? Mainly for consistency with the node.

File: wallet/src/repository/wallet/mod.rs [hunk 5]
```diff
@@ -380,16 +382,111 @@ where
...
+
+        Ok(transaction)
+    }
+
+    /// Create a new data request transaction using available UTXOs.
+    pub fn create_data_req(
+        &self,
+        types::DataReqParams { label, request }: types::DataReqParams,
+    ) -> Result<types::DRTransaction> {
+        let fee = request.fee();
```
Comment (tmpolaczyk) [inline wallet/src/repository/wallet/mod.rs]: This is not the fee we want to use here, it should be a new parameter.

File: wallet/src/repository/wallet/mod.rs [hunk 6]
```diff
@@ -409,51 +418,143 @@ where
...
+    /// Create a new data request transaction using available UTXOs.
+    pub fn create_data_req(
+        &self,
+        types::DataReqParams {
+            label,
+            fee,
+            request,
+        }: types::DataReqParams,
+    ) -> Result<types::DRTransaction> {
+        let value = request.total_witnesses_reward();
```
Comment (tmpolaczyk) [inline wallet/src/repository/wallet/mod.rs]:
```suggestion
        let value = request.value;
```

File: wallet/src/repository/wallet/mod.rs [hunk 7]
```diff
@@ -599,7 +700,7 @@ where
             batch.put(&keys::transaction_value(account, txn_id), amount)?;
             batch.put(
                 keys::transaction_type(account, txn_id),
```
Comment (tmpolaczyk) [inline wallet/src/repository/wallet/mod.rs]:
```suggestion
                keys::transaction_entry(account, txn_id),
```

File: wallet/src/repository/wallet/mod.rs [hunk 8]
```diff
@@ -527,7 +628,7 @@ where
...
```
Comment (tmpolaczyk) [inline wallet/src/repository/wallet/mod.rs]:
```suggestion
                keys::transaction_entry(account, txn_id),
```

**Pull request merged**